src/api/v1.py

```diff
@@ -343,6 +343,33 @@ def images_post():
     return "OK"
 
 
+def agents_post(agent_name):
+    """Post information about the agent to the server
+
+    The json sent to this endpoint may contain data such as the following:
+    {
+        "state": string, # State the device is in
+        "queues": array[string], # Queues the device is listening on
+        "location": string, # Location of the device
+        "job_id": string, # Job ID the device is running, if any
+        "log": array[string], # push and keep only the last 100 lines
+    }
+    """
+
+    data = request.get_json()
+    data["name"] = agent_name
```

Comment:
Shouldn't we add a verification that the posted data is correct and if it’s not, return the 400  status code?

Reply:
In this case, we don't always know what we're going to get. The agent is just self-reporting some data about it so that we can reflect that from the server side. Sometimes it could be telling us information like the queues it listens on, other times it could be updating the latest logs. Or is there another situation you are concerned about?

Reply:
hmm, yeah actually I guess if we sent it something other than json, it could wind up getting a string or something back from get_json(). Let's just check and ensure it's a dict before adding things to it, good idea!

Reply:
I mean, what if the JSON data won't have "name" key? In this case, you'll get a KeyError and a client will get 500 status code. However, it's not a server mistake, but the client's one. So, I guess we should check is the key is present in the data, and if it's not, we should respond with 400 code.

Reply:
I mean, what if the JSON data won't have "name" key? In this case, you'll get a KeyError and a client will get 500 status code. However, it's not a server mistake, but the client's one. So, I guess we should check is the key is present in the data, and if it's not, we should respond with 400 code.

Reply:
Could you please add the check that the “name” and “updated_at” keys exist in the data?

Reply:
They should not normally exist in the data that we get from get_json(). If they do, then they will be replaced. The idea here is that we will add those keys to the dict before pushing the record to the db, along with the other data sent by the agent.

Reply:
Okay, thanks!

Reply:
_the missing comment:_ I mean, what if the JSON data won't have "name" key? In this case, you'll get a KeyError and a client will get 500 status code. However, it's not a server mistake, but the client's one. So, I guess we should check is the key is present in the data, and if it's not, we should respond with 400 code.